app/services/the_tvdb_service.rb

```diff
@@ -0,0 +1,124 @@
+class TheTvdbService
+  BASE_URL = 'https://api.thetvdb.com'.freeze
+  API_KEY = ENV['THE_TVDB_API_KEY'].freeze
+
+  attr_accessor :media_id
+
+  def initialize
+    @token = api_token
+  end
+
+  # This will update all mappings to have a season number
+  # This should be run ONLY ONCE, unless we import data from somewhere else.
+  def update_mapping_seasons!
+    return if mapping_data.blank?
+
+    mapping_data['Anime'].each do |media_ids|
+      # set the media_id (this is for our database)
+      self.media_id = media_ids[0]
+      series_id = media_ids[1]['thetvdb/series'][1]
+      season_id = media_ids[1]['thetvdb/season'][1] if media_ids[1].try(:[], 'thetvdb/season')
```

Comment:
Wouldn't it make more sense to destructure this in the parameters like `|(anime_id, mappings)|` and then use `mappings['thetvdb/series']`

Reply:
not sure I follow, but I realized the `.try` was pointless.